src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1523,7 +1563,13 @@ void ImportInitBlk()
 
         void ImportRethrow()
         {
-            // TODO:
+            if (_currentBasicBlock.HandlerIndex.HasValue &&
+             _exceptionRegions[_currentBasicBlock.HandlerIndex.Value].ILRegion.Kind == ILExceptionRegionKind.Catch)
```

Comment:
This will incorrectly fail in the handler for filter. E.g. the following is good IL, but this will fail for it:

```
  .method private hidebysig static void  Main() cil managed
  {
    .entrypoint
    .maxstack  1
    .try
    {
       leave.s    Done
    }  // end .try
    filter
    {
       ldc.i4.0
       endfilter
    }  // end filter
    {  // handler
       rethrow // this should be allowed
    }  // end handler
Done:
    ret
  } // end of method My::Main
```